### PR TITLE
Store information about partial ngen in R2R images

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
@jkotas @sergiy-k this adds the option discussed in email.

### CODE REVIEW

File: src/inc/readytorun.h [hunk 1]
```diff
@@ -45,6 +45,7 @@ enum ReadyToRunFlag
     // Set if the original IL assembly was platform-neutral
     READYTORUN_FLAG_PLATFORM_NEUTRAL_SOURCE         = 0x00000001,
     READYTORUN_FLAG_SKIP_TYPE_VALIDATION            = 0x00000002,
+    READYTORUN_FLAG_PARTIAL_NGEN                    = 0x00000004,
```
Comment (jkotas) [inline src/inc/readytorun.h]: I would not include `NGen` in the name of R2R flag. What about:

```
// Set of methods with native code was determined using profile data 
READYTORUN_FLAG_PARTIAL = ...,
```

**Comment by AndyAyersMS** [comment]
You might consider exposing this via ETW too, perhaps via some new flag here:

https://github.com/dotnet/coreclr/blob/88249918cd01e8729c60c0f6c2cf8145d0bc3c55/src/inc/eventtracebase.h#L526-L534

File: src/inc/eventtracebase.h [hunk 2]
```diff
@@ -531,6 +531,7 @@ namespace ETW
                 ManifestModule=0x8,
                 IbcOptimized=0x10,
                 ReadyToRunModule=0x20,
+                PartialReadyToRunModule=0x40,
```
Comment (jkotas) [inline src/inc/eventtracebase.h]: @vancem Should this flag be listed in ClrEtwAll.man as well?

Looks like a few of the recently added flags are missing in ClrEtwAll.man.

Comment (MichalStrehovsky) [reply src/inc/eventtracebase.h]: Is this blocking or can we just file an issue to look what we need to do for values `> 0x8` and merge this?

Comment (jkotas) [reply src/inc/eventtracebase.h]: It is ok to file an issue.

Comment (MichalStrehovsky) [reply src/inc/eventtracebase.h]: Thanks! Filed #22790 in the Tracing area path for 3.0.

Comment (vancem) [reply src/inc/eventtracebase.h]: > @vancem Should this flag be listed in ClrEtwAll.man as well?

The answer is yes, I will update the issue with more.

I have already updated PerfVIew's enum for this so you will see it parsed in the 'events' view in PerfView

**Pull request merged**